### ISSUE TITLE
A check fail can be triggered in MatrixSolve

### ISSUE THREAD

**Issue by shijy16** [tensorflow/tensorflow#60278]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

binary

### Tensorflow Version

2.13.0-dev20230406

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.10

### CUDA/cuDNN version

CUDA 11.8

### Current Behaviour?

```shell
The following code can trigger a crash in `tf.raw_ops.MatrixSolve` due to check-fail(multiply overflow) in the latest version of TensorFlow.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
with tf.device("CPU"):
    adjoint = False
    matrix = tf.random.uniform([0, 3, 15, 7847250026211090813, 11], dtype=tf.float64, minval=-1024, maxval=1024)
    rhs = tf.random.uniform([11], dtype=tf.float64, minval=-1024, maxval=1024)
    res = tf.raw_ops.MatrixSolve(
        adjoint=adjoint,
        matrix=matrix,
        rhs=rhs,
    )
```

### Relevant log output

```shell
2023-04-09 14:04:15.420939: I tensorflow/core/util/port.cc:110] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-04-09 14:04:15.468322: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-04-09 14:04:16.265727: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2023-04-09 14:04:17.807591: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1638] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 14561 MB memory:  -> device: 0, name: Tesla V100-PCIE-16GB, pci bus id: 0000:2f:00.0, compute capability: 7.0
2023-04-09 14:04:17.844083: F tensorflow/core/framework/tensor_shape.cc:201] Non-OK-status: InitDims(dim_sizes) status: INVALID_ARGUMENT: Encountered overflow when multiplying 7847250026211090813 with 11, result: -1
Aborted (core dumped)
```
</details>

**Comment by tiruk007** [comment]
@SuryanarayanaY 
I was able to reproduce the issue on Colab using tf 2.12 and tf-nightly. Please find the gist of [2.12](https://colab.research.google.com/gist/tiruk007/d6c87adf711369429ae38ead62a7d322/untitled182.ipynb) and [tf-nightly](https://colab.research.google.com/gist/tiruk007/9f02ac71d52e640f864f0736c59fed97/tf-nightly.ipynb#scrollTo=Az7CQKKwobxM) for reference.

**Comment by SuryanarayanaY** [comment]
Hi @shijy16 ,

The issue got resolved now with tf-nightly(2.14.0-dev20230514) and now TF is successfully able to raise the intended Error.
Please refer to attached [gist](https://colab.research.google.com/gist/SuryanarayanaY/11f2a25b393d2532bf781b08b3849ff6/60278.ipynb).

Please check and confirm and let us know if we can close the issue as it resolved now.

Thanks!
